### PR TITLE
CloudDNS swagger updates

### PR DESCRIPTION
**Pull request by vladrai**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 
### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).
### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [ ] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

---

This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/615)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vladrai__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (vladrai). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: arm-dns/2016-04-01/swagger/dns.json [hunk 1]
```diff
@@ -431,8 +430,9 @@
             "name": "$top",
             "in": "query",
             "required": false,
-            "type": "string",
-            "description": "Query parameters. If null is passed returns the default number of zones."
+            "type": "integer",
+            "format": "int32",
+            "description": "Query parameters. If not specified returns the default number of recordsets."
```
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: and what is that default value (number of recordsets)?

File: arm-dns/2016-04-01/swagger/dns.json [hunk 2]
```diff
@@ -486,8 +486,9 @@
             "name": "$top",
             "in": "query",
             "required": false,
-            "type": "string",
-            "description": "Query parameters. If null is passed returns the default number of zones."
+            "type": "integer",
+            "format": "int32",
+            "description": "Query parameters. If not specified returns the default number of zones."
```
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: same as above

File: arm-dns/2016-04-01/swagger/dns.json [hunk 3]
```diff
@@ -719,8 +720,9 @@
...
```
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: same as above.

File: arm-dns/2016-04-01/swagger/dns.json [hunk 4]
```diff
@@ -753,15 +755,16 @@
...
         "parameters": [
           {
             "name": "$top",
             "in": "query",
             "required": false,
-            "type": "string",
-            "description": "Query parameters. If null is passed returns the default number of zones."
+            "type": "integer",
+            "format": "int32",
+            "description": "Query parameters. If not specified returns the default number of zones."
```
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: same as above.

**Comment by amarzavery** [comment]
@anuchandy - Please make sure that all the comments provided in the email have been addressed.

**Comment by vladrai** [comment]
Something wrong with build system?

ERROR:__main__:Command '['mono', '/tmp/tmps41xcaxw/autorest/tools/AutoRest.exe', '-i', '/git-restapi/arm-dns/2016-04-01/swagger/dns.json', '-o', '/tmp/tmps41xcaxw/dns.json', '-AddCredentials', 'True', '-CodeGenerator', 'Azure.Python', '-Header', 'MICROSOFT_MIT_NO_VERSION', '-Namespace', 'azure.mgmt.dns', '-PackageVersion', '0.30.0rc6', '-ft', '2']' returned non-zero exit status 1
ERROR:__main__:Code generation failed.
AutoRest.exe -i /git-restapi/arm-dns/2016-04-01/swagger/dns.json -o /tmp/tmps41xcaxw/dns.json -AddCredentials True -CodeGenerator Azure.Python -Header MICROSOFT_MIT_NO_VERSION -Namespace azure.mgmt.dns -PackageVersion 0.30.0rc6 -ft 2
FATAL: Error loading Azure.Python assembly: Could not load file or assembly 'AutoRest.Python.Azure.dll' or one of its dependencies

**Comment by vladrai** [comment]
I addressed all comments and build passed this time. Please merge.

**Comment by tbombach** [comment]
Hi @vladrai - it seems like you're making improvements to documentation. Have you engaged with folks on the docs team? @erickson-doug has provided guidance here: https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-authoring-descriptions.md and I want to make sure you've been given that resource if your changes are intended for the API reference docs.

File: arm-dns/2016-04-01/swagger/dns.json (hunk 1 above)
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: @vladrai - What is that default number? Is it possible to specify that value in the description?

File: arm-dns/2016-04-01/swagger/dns.json (hunk 4 above)
Comment (amarzavery) [inline arm-dns/2016-04-01/swagger/dns.json]: I see thanks. Ignore the comment that I just made today.

**Pull request merged**